src/lib/tls/tls13_pqc/hybrid_public_key.cpp

```diff
@@ -349,6 +349,12 @@ Hybrid_KEM_PrivateKey::Hybrid_KEM_PrivateKey(std::vector<std::unique_ptr<Private
       });
 }
 
+Hybrid_KEM_PrivateKey& Hybrid_KEM_PrivateKey::operator=(Hybrid_KEM_PrivateKey&& other) {
+   static_cast<Hybrid_KEM_PublicKey&>(*this) = static_cast<Hybrid_KEM_PublicKey&&>(other);
```

Comment:
That looks somewhat unusual. Could paste the warning clangcl did generate for the original code please?

Reply:
```
build/include/public\botan/dh.h(80,30): error: defaulted move assignment operator of 'DH_PrivateKey' will move assign virtual base class 'Private_Key' multiple times [-Werror,-Wmultiple-move-vbase]
   80 | class BOTAN_PUBLIC_API(2, 0) DH_PrivateKey final : public DH_PublicKey,
      |                              ^
build/include/public\botan/dh.h(80,7): note: due to 'DH_PrivateKey' being dllexported
   80 | class BOTAN_PUBLIC_API(2, 0) DH_PrivateKey final : public DH_PublicKey,
      |       ^
build/include/public\botan/compiler.h(31,36): note: expanded from macro 'BOTAN_PUBLIC_API'
   31 | #define BOTAN_PUBLIC_API(maj, min) BOTAN_DLL
      |                                    ^
<command line>(3,30): note: expanded from macro 'BOTAN_DLL'
    3 | #define BOTAN_DLL __declspec(dllexport)
      |                              ^
build/include/public\botan/dh.h(81,52): note: 'Private_Key' is a virtual base class of base class 'PK_Key_Agreement_Key' declared here
   81 |                                                    public PK_Key_Agreement_Key,
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
build/include/public\botan/dh.h(82,52): note: virtual base class 'Private_Key' declared here
   82 |                                                    public virtual Private_Key {
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
I think, instead of defining custom move constructors as a workaround, we actually want to add `virtual` inheritance to a few more base class declarations. See this patch: https://github.com/reneme/botan/commit/f663994ad84c62127d419b26c1787e5e9e016805

Reply:
Much better!